### ISSUE TITLE
Block scope collision checking is broken

### ISSUE THREAD

**Issue by daliwali** [babel/babel#1130]
This code worked before:

```js
function A () { let a }
function B () { let a }
```

Now I get a TypeError:

```
TypeError: foo.js: Line 2: Duplicate declaration "a"
  1 | function A () { let a }
> 2 | function B () { let a }
    |                     ^
  3 | 
    at File.errorWithNode (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/transformation/file/index.js:436:15)
    at Scope.checkBlockScopedCollisions (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/scope.js:287:23)
    at Scope.registerBinding (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/scope.js:424:12)
    at Scope.registerDeclaration (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/scope.js:384:14)
    at TraversalPath.enter (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/scope.js:79:13)
    at TraversalPath.call (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/path/index.js:443:26)
    at TraversalPath.visit (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/path/index.js:455:10)
    at TraversalContext.visitMultiple (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/context.js:50:16)
    at TraversalContext.visit (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/context.js:71:19)
    at Function.traverse.node (/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/traversal/index.js:44:17)

```

Is this a bug, or did block scoping change, or am I going crazy?

**Comment by sebmck** [comment]
Shit, definently a bug.

**Comment by sebmck** [comment]
I'm currently pushing out 5.0.0. After I'm doing updating all the plugins I'll push out a patch release.

**Comment by sebmck** [comment]
Fixed as of 5.0.2, thanks!

**Comment by daliwali** [comment]
That was fast! :+1:
